### PR TITLE
snap_alias - new module

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
New module `snap_alias` to manage aliases for installed `snap`s.

Fixes: #566

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
snap_alias

### CODE REVIEW

File: plugins/modules/packaging/os/snap_alias.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+- name: Remove all aliases for snap
+  community.general.snap_alias:
+    name: hello-world
+    state: absent
+'''
+
+RETURN = '''
+snap_aliases:
+    description: The snap aliases after execution
+    type: list
```
Comment (felixfontein) [inline plugins/modules/packaging/os/snap_alias.py:66]:
```suggestion
    type: list
    elements: str
```

File: tests/integration/targets/snap_alias/tasks/test.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+- name: Ensure snap 'hello-world' is installed now
+  community.general.snap:
+    name: hello-world
+
+- name: Create snap alias
+  community.general.snap_alias:
+    name: hello-world
+    alias: hw
+  register: alias_single
```
Comment (felixfontein) [inline tests/integration/targets/snap_alias/tasks/test.yml:15]: How about also testing check mode and idempotency? I.e. replace this one task by four ones (check mode, regular, check mode idempotency, regular idempotency). (Same then for the other tasks.)

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I don't know much about snap, so more some general comments :)

File: plugins/modules/packaging/os/snap_alias.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    alias:
+        description:
+            - Aliases to be created or removed.
+        type: list
+        elements: str
+        aliases: [aliases]
+
+author:
+    - Alexei Znamensky (@russoz) <russoz@gmail.com>
+'''
```
[the remark this answers is not in the document]
Comment (russoz) [reply plugins/modules/packaging/os/snap_alias.py:38]: done, coming up soon

File: tests/integration/targets/snap_alias/tasks/test.yml (hunk 2 above)
Comment (russoz) [reply tests/integration/targets/snap_alias/tasks/test.yml:15]: supporting check mode would be a good start :P

Comment (felixfontein) [reply tests/integration/targets/snap_alias/tasks/test.yml:15]: In that case, you should mention in `notes` that check_mode is not supported.

Comment (felixfontein) [reply tests/integration/targets/snap_alias/tasks/test.yml:15]: (And idempotency checks still make sense :) )

**Comment by russoz** [comment]
> In that case, you should mention in `notes` that check_mode is not supported. 

<voice type="darthvader">I find your lack of faith disturbing, Admiral</voice>. I am halfway through implementing check mode. ;-)

File: tests/integration/targets/snap_alias/tasks/test.yml (hunk 2 above)
Comment (russoz) [reply tests/integration/targets/snap_alias/tasks/test.yml]: done

File: plugins/modules/packaging/os/snap_alias.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+- name: Remove all aliases for snap
+  community.general.snap_alias:
+    name: hello-world
+    state: absent
+'''
+
+RETURN = '''
+snap_aliases:
+    description: The snap aliases after execution
```
Comment (felixfontein) [inline plugins/modules/packaging/os/snap_alias.py:72]:
```suggestion
    description: The snap aliases after execution.
```

File: plugins/modules/packaging/os/snap_alias.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+    type: list
+    elements: str
+    returned: always
```
Comment (felixfontein) [inline plugins/modules/packaging/os/snap_alias.py:75]: Also in check mode? (Because you're not checking that in the check mode tests.)

Comment (russoz) [reply plugins/modules/packaging/os/snap_alias.py:75]: In check mode the list reflects the "before" state. Just updated the description with that info.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for contributing this!

**Pull request merged**